_studio/mfx_lib/decode/av1/src/mfx_av1_dec_decode.cpp

```diff
@@ -0,0 +1,1191 @@
+            surface_out->Data.Corrupted |= MFX_CORRUPTION_REFERENCE_FRAME;
+
+        if (error & UMC::ERROR_FRAME_DPB)
+            surface_out->Data.Corrupted |= MFX_CORRUPTION_REFERENCE_LIST;
+
+        if (error & UMC::ERROR_FRAME_RECOVERY)
+            surface_out->Data.Corrupted |= MFX_CORRUPTION_MAJOR;
+
+        if (error & UMC::ERROR_FRAME_TOP_FIELD_ABSENT)
+            surface_out->Data.Corrupted |= MFX_CORRUPTION_ABSENT_TOP_FIELD;
+
+        if (error & UMC::ERROR_FRAME_BOTTOM_FIELD_ABSENT)
+            surface_out->Data.Corrupted |= MFX_CORRUPTION_ABSENT_BOTTOM_FIELD;
+    }
+
+    UMC::FrameMemID id = frame->GetFrameData()->GetFrameMID();
+    mfxStatus sts = m_allocator->PrepareToOutput(surface_out, id, &m_video_par, m_opaque);
+#ifdef MFX_VA
+    frame->Displayed(true);
+#else
+    frame->Reset();
+#endif
+
+    return sts;
+}
+
+mfxStatus VideoDECODEAV1::QueryFrame(mfxThreadTask task)
+{
+    MFX_CHECK_NULL_PTR1(task);
+
+    MFX_CHECK(m_core, MFX_ERR_UNDEFINED_BEHAVIOR);
+    MFX_CHECK(m_decoder, MFX_ERR_NOT_INITIALIZED);
+
+    auto info =
+        reinterpret_cast<TaskInfo*>(task);
+
+    mfxFrameSurface1* surface_out = info->surface_out;
+    MFX_CHECK(surface_out, MFX_ERR_UNDEFINED_BEHAVIOR);
+
+    UMC::FrameMemID id = m_allocator->FindSurface(surface_out, m_opaque);
+    UMC_AV1_DECODER::AV1DecoderFrame* frame =
+        m_decoder->FindFrameByMemID(id);
+
+    MFX_CHECK(frame, MFX_ERR_UNDEFINED_BEHAVIOR);
+
+    MFX_CHECK(frame->DecodingStarted(), MFX_ERR_UNDEFINED_BEHAVIOR);
+
+    if (!frame->DecodingCompleted())
+    {
+        m_decoder->QueryFrames();
+    }
+
+    MFX_CHECK(frame->DecodingCompleted(), MFX_TASK_WORKING);
+
+    mfxStatus sts = DecodeFrame(surface_out, frame);
+    MFX_CHECK_STS(sts);
+
+    return MFX_TASK_DONE;
+}
+
+static mfxStatus CheckFrameInfo(mfxFrameInfo const &currInfo, mfxFrameInfo &info)
+{
+    MFX_SAFE_CALL(CheckFrameInfoCommon(&info, MFX_CODEC_AV1));
+
+    switch (info.FourCC)
+    {
+    case MFX_FOURCC_NV12:
+    case MFX_FOURCC_AYUV:
+#if (MFX_VERSION >= 1027)
+    case MFX_FOURCC_Y410:
+#endif
+        break;
+#if (MFX_VERSION >= 1330)
+    case MFX_FOURCC_P016:
+    case MFX_FOURCC_Y416:
+#endif
+    case MFX_FOURCC_P010:
+#if (MFX_VERSION >= 1027)
+    case MFX_FOURCC_Y210:
+#endif
+        MFX_CHECK(info.Shift == 1, MFX_ERR_INCOMPATIBLE_VIDEO_PARAM);
+        break;
+    default:
+        MFX_CHECK_STS(MFX_ERR_INVALID_VIDEO_PARAM);
+    }
+
+    switch (info.ChromaFormat)
+    {
+    case MFX_CHROMAFORMAT_YUV420:
+    case MFX_CHROMAFORMAT_YUV444:
+        break;
+    default:
+        MFX_CHECK_STS(MFX_ERR_INVALID_VIDEO_PARAM);
+    }
+
+    MFX_CHECK(currInfo.FourCC == info.FourCC, MFX_ERR_INCOMPATIBLE_VIDEO_PARAM);
+
+    MFX_CHECK(info.Width <= currInfo.Width && info.Height <= currInfo.Height, MFX_ERR_INCOMPATIBLE_VIDEO_PARAM);
+
+    MFX_CHECK((info.CropW && info.Width >= info.CropW) || (info.CropH && info.Height >= info.CropH), MFX_ERR_INCOMPATIBLE_VIDEO_PARAM);
```

Comment:
I didn't see the above checks for other Codecs.
E.g. vp9 decode
https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/mfx_lib/decode/vp9/src/mfx_vp9_dec_decode_hw.cpp#L906-L944

Could you remove line 786-788 to keep the same behavior with other CODECs?

Reply:
Sure. I will remove it.